expression/simple_rewriter.go

```diff
@@ -27,26 +27,28 @@ import (
 type simpleRewriter struct {
 	exprStack
 
-	tbl *model.TableInfo
-	err error
-	ctx sessionctx.Context
+	schema *Schema
+	err    error
+	ctx    sessionctx.Context
 }
 
-// ParseSimpleExpr parses simple expression string to Expression.
+// ParseSimpleExprWithTableInfo parses simple expression string to Expression.
 // The expression string must only reference the column in table Info.
-func ParseSimpleExpr(ctx sessionctx.Context, exprStr string, tableInfo *model.TableInfo) (Expression, error) {
+func ParseSimpleExprWithTableInfo(ctx sessionctx.Context, exprStr string, tableInfo *model.TableInfo) (Expression, error) {
```

Comment:
If we don't have a function named `ParseSimpleExprWithoutTableInfo`, I think `ParseSimpleExpr` is a better name, it's shorter.

We can change the parameter ordinal in this way:
```go
ParseSimpleExprWithTableInfo(ctx sessionctx.Context, tableInfo *model.TableInfo, exprStr string)
RewriteSimpleExprWithTableInfo(ctx sessionctx.Context, tbl *model.TableInfo, expr ast.ExprNode)
```

Reply:
+1
We don't need to change the name.